v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/uniformsplitter/range/BoundarySplitterFactory.java

```diff
@@ -172,6 +177,37 @@ private static BigDecimal splitBigDecimal(BigDecimal start, BigDecimal end) {
     return new BigDecimal(split);
   }
 
+  private static Date splitDates(Date start, Date end) {
+    if (start == null && end == null) {
+      return null;
+    }
+    if (start == null) {
+      start = Date.valueOf(LocalDate.MIN);
+    }
+    if (end == null) {
+      end = Date.valueOf(LocalDate.MIN);
+    }
+
+    int startDateInt = convertDateToInt(start);
+    int endDateInt = convertDateToInt(end);
+
+    int dateMid = splitIntegers(startDateInt, endDateInt);
+
+    return convertIntToSqlDate(dateMid);
+  }
+
+  private static int convertDateToInt(Date sqlDate) {
+    LocalDate localDate = sqlDate.toLocalDate();
+    String dateString = localDate.format(DateTimeFormatter.BASIC_ISO_DATE);
```

Comment:
[BASIC_ISO_DATE](https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#BASIC_ISO_DATE) is not naturally splittable. It's more of a integer parsable stringified representation of the date.

Would it make more sense to convert it to number of days post epoch?